src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerFindMatches.java

```diff
@@ -83,39 +85,58 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToStart(final CaseDeta
 
     }
 
-    // Helper method to create term queries with variations for spaces
-    private BoolQueryBuilder createNameMatchQuery(String field, String name) {
-        return QueryBuilders.boolQuery()
-            .should(QueryBuilders.termQuery(field, name)) // Exact match
-            .should(QueryBuilders.termQuery(field, " " + name)) // Prepend space
-            .should(QueryBuilders.termQuery(field, name + " ")) // Append space
-            .should(QueryBuilders.termQuery(field, " " + name + " ")); // Prepend and append space
+    public String[] normalizeAndSplit(String name) {
+        // remove brackets and anything inside them
+        String nameWithoutStuffInBrackets = name.replaceAll("\\([^)]*\\)", "");
+
+        // replace multiple consecutive "/" with a single "/" because we might have multiple names due to translations
+        String nameWithoutExtraSlashes = nameWithoutStuffInBrackets.replaceAll("/+", "/");
```

Comment:
It's difficult to know what's best for this, as other characters might be used to separate translations 🤔. I was thinking it could make sense to remove all punctuation and then we can use a simple regex (`[^a-zA-Z0-9]`). Not sure whether that will cater to every scenario though

Reply:
If I remove all punctuation I'm not going to be able to match on the other names they might have given.
Sion Corn / Santa Claus would then be Sion Corn Santa Claus so wouldn't match on either Sion Corn nor Santa Claus I think. 
One thing I did wonder about was splitting off according to any weird characters and then doing the matches on the shorter names - so might be doing match on first name, last name, foreign spelling first name, foreign spelling last name and it would give more false positives but all on matching marriage date only.
Then this example would match on Sion, Corn, Santa, Claus and also Sion Corn, Sion Siabod, Sian Sion etc.
What do you reckon?

Reply:
we are going with splitting stored name to array of names where there are illegal chars but also removing 'name changed by deed poll'